modules/silicon/main.tf

```diff
@@ -0,0 +1,222 @@
+}
+
+module "vpc_ai_notebook" {
+  count   = var.create_network ? 1 : 0
+  source  = "terraform-google-modules/network/google"
+  version = "~> 3.0"
+
+  project_id   = local.project.project_id
+  network_name = var.network_name
+  routing_mode = "GLOBAL"
+  description  = "VPC Network created via Terraform"
+
+  subnets = [
+    {
+      subnet_name           = var.subnet_name
+      subnet_ip             = var.ip_cidr_range
+      subnet_region         = local.region
+      description           = "Subnetwork inside *vpc-silicon* VPC network, created via Terraform"
+      subnet_private_access = true
+    }
+  ]
+
+  firewall_rules = [
+    {
+      name        = "fw-ai-notebook-allow-internal"
+      description = "Firewall rule to allow traffic on all ports inside *vpc-silicon* VPC network."
+      priority    = 65534
+      ranges      = ["10.0.0.0/8"]
+      direction   = "INGRESS"
+
+      allow = [{
+        protocol = "tcp"
+        ports    = ["0-65535"]
+      }]
+    }
+  ]
+
+  depends_on = [
+    google_project_service.enabled_services
+  ]
+}
+
+resource "google_service_account" "sa_p_notebook" {
+  project      = local.project.project_id
+  account_id   = format("sa-p-notebook-%s", local.random_id)
+  display_name = "Notebooks in trusted environment"
+}
+
+resource "google_project_iam_member" "sa_p_notebook_permissions" {
+  for_each = toset(local.notebook_sa_project_roles)
+  project  = local.project.project_id
+  member   = "serviceAccount:${google_service_account.sa_p_notebook.email}"
+  role     = each.value
+}
+
+resource "google_service_account_iam_member" "sa_ai_notebook_user_iam" {
+  for_each           = var.trusted_users
+  member             = each.value
+  role               = "roles/iam.serviceAccountUser"
+  service_account_id = google_service_account.sa_p_notebook.id
+}
+
+resource "google_project_iam_binding" "ai_notebook_user_role1" {
+  project = local.project.project_id
+  members = var.trusted_users
+  role    = "roles/notebooks.admin"
+}
+
+resource "google_project_iam_binding" "ai_notebook_user_role2" {
+  project = local.project.project_id
+  members = var.trusted_users
+  role    = "roles/viewer"
+}
+
+resource "google_notebooks_instance" "ai_notebook" {
+  count        = var.notebook_count
+  project      = local.project.project_id
+  name         = "notebooks-instance-${count.index}"
+  location     = var.zone
+  machine_type = var.machine_type
+
+   container_image {
+    repository = "gcr.io/${local.project.project_id}/openlane-jupyterlab"
+    tag = "latest"
+   }
+
+  service_account = google_service_account.sa_p_notebook.email
+
+  install_gpu_driver = false
+  boot_disk_type     = var.boot_disk_type
+  boot_disk_size_gb  = var.boot_disk_size_gb
+
+  no_public_ip    = false
+  no_proxy_access = false
+
+  network = local.network.self_link
+  subnet  = local.subnet.self_link
+
+  labels = {
+    module = "silicon"
+  }
+
+  metadata = {
+    terraform  = "true"
+    proxy-mode = "mail"
```

Comment:
according to https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/notebooks_instance#example-usage---notebook-instance-basic-container this should probably be `service_account`.

Reply:
Done